dimensiondata/src/test/java/org/jclouds/dimensiondata/cloudcontrol/features/ServerApiLiveTest.java

```diff
@@ -58,65 +90,223 @@ public void testListServers() {
+      assertTrue(!natRulesList.isEmpty());
+      assertEquals(natRulesList.last().get().first().get().networkDomainId(), networkDomainId);
+   }
+
+   @Test(dependsOnMethods = { "testCreateNatRule", "testListNatRules" })
+   public void testGetNatRule() {
+      NatRule natRule = api.getNetworkApi().getNatRule(natRuleId);
+      assertNotNull(natRule);
+      assertEquals(natRule.networkDomainId(), networkDomainId);
+   }
+
+   @Test(dependsOnMethods = "testGetNatRule", alwaysRun = true)
+   public void testDeleteNatRule() {
+      api.getNetworkApi().deleteNatRule(natRuleId);
+      NatRule natRule = api.getNetworkApi().getNatRule(natRuleId);
+      assertNull(natRule);
+   }
+
+   @Test(dependsOnMethods = { "testDeleteNatRule" })
+   public void testRemovePublicIpBlock() {
+      api.getNetworkApi().removePublicIpBlock(publicIpv4BlockId);
+      publicIpBlock = api.getNetworkApi().getPublicIPv4AddressBlock(publicIpv4BlockId);
+      assertNull(publicIpBlock);
+   }
+
    @Test(dependsOnMethods = "testRebootServer")
    public void testPowerOffServer() {
-      api().powerOffServer(serverId);
+      api.getServerApi().powerOffServer(serverId);
       assertTrue(serverStoppedPredicate.apply(serverId), "server did not power off after timeout");
    }
 
    @Test(dependsOnMethods = "testPowerOffServer")
    public void testStartServer() {
-      api().startServer(serverId);
+      api.getServerApi().startServer(serverId);
       assertTrue(serverStartedPredicate.apply(serverId), "server did not start after timeout");
       assertTrue(vmtoolsRunningPredicate.apply(serverId), "server vm tools not running after timeout");
    }
 
    @Test(dependsOnMethods = "testStartServer")
    public void testShutdownServer() {
-      api().shutdownServer(serverId);
+      api.getServerApi().shutdownServer(serverId);
       assertTrue(serverStoppedPredicate.apply(serverId), "server did not shutdown after timeout");
    }
 
    @Test(dependsOnMethods = "testShutdownServer")
-   public void testCloneServer() {
+   public void testCloneServerToMakeCustomerImage() {
       CloneServerOptions options = CloneServerOptions.builder().clusterId("").description("")
             .guestOsCustomization(false).build();
-      cloneImageId = api().cloneServer(serverId, "ServerApiLiveTest-" + System.currentTimeMillis(), options);
+      cloneImageId = api.getServerApi()
+            .cloneServer(serverId, "ServerApiLiveTest-" + System.currentTimeMillis(), options);
       assertNotNull(cloneImageId);
       assertTrue(serverNormalPredicate.apply(serverId), "server was not NORMAL after timeout");
    }
 
+   @Test(dependsOnMethods = "testCloneServerToMakeCustomerImage")
+   public void testListCustomerImages() {
+      FluentIterable<CustomerImage> customerImages = api.getServerImageApi().listCustomerImages().concat();
+      assertNotNull(customerImages);
+      customerImages.anyMatch(new Predicate<CustomerImage>() {
```

Comment:
`assertTrue(customerImages.anyMatch(new Predicate<CustomerImage>() {`

Reply:
Thanks for spotting that one! Not much point in having the predicate without the assertion.